### PR TITLE
syncer: support sync table or db

### PR DESCRIPTION
**Pull request by nolouch**
- support select table
- support select database
PTAL @siddontang  @qiuyesuifeng 

close https://github.com/pingcap/tidb-tools/issues/23

### CODE REVIEW

**Comment by andelf** [comment]
Feature request: replicate-ignore-db & replicate-ignore-table.

File: syncer/config.toml [hunk 1]
```diff
@@ -10,6 +10,18 @@ batch = 1
 
 pprof-addr = ":10081"
 
+#replicate-do-db priority over replicate-do-tableif have same db name
```
Comment (qiuyesuifeng) [inline syncer/config.toml]: I prefer to use regular rule for replication, how about you? @siddontang

**Comment by nolouch** [comment]
@andelf @qiuyesuifeng @GregoryIan @siddontang PTAL

File: syncer/config.go [hunk 2]
```diff
@@ -78,6 +86,10 @@ type Config struct {
 
 	Meta string `toml:"meta" json:"meta"`
 
+	DoTable []TableName `toml:"replicate-do-table" json:"replicate-do-table"`
```
Comment (siddontang) [inline syncer/config.go]: what is `Do` here?

Comment (nolouch) [inline syncer/config.go]: I found mysql argument named `replicate-do-table` so named `DoTable` here.

File: syncer/syncer.go [hunk 3]
```diff
@@ -458,7 +468,85 @@ func (s *Syncer) sync(db *sql.DB, jobChan chan *job) {
 	}
 }
 
-func (s *Syncer) skip(sql string) bool {
+func (s *Syncer) matchDB(patternDBS []string, a string) bool {
+	for _, b := range patternDBS {
+		if re, ok := s.reMap[b]; ok {
```
Comment (GregoryIan) [inline syncer/syncer.go]: u can use `(s *Syncer) matchString(pattern string, t string)`

File: syncer/syncer.go [hunk 4]
```diff
@@ -641,6 +768,30 @@ func (s *Syncer) run() error {
...
+		if db[0] != '~' {
+			continue
+		}
+		if _, ok := s.reMap[db]; !ok {
+			s.reMap[db] = regexp.MustCompile(db[1:])
+		}
+	}
+
+	for _, tb := range s.cfg.DoTable {
+		if tb.Name[0] != '~' {
```
Comment (GregoryIan) [inline syncer/syncer.go]: add `|| tb.Schema[0] != '~'`

File: syncer/config.toml [hunk 5]
```diff
@@ -4,12 +4,25 @@ log-level = "info"
...
 batch = 1
 
 pprof-addr = ":10081"
 
+##replicate-do-db priority over replicate-do-table if have same db name
+##and we support regex expression , start with '~' declare use regex expression.
+#
+#replicate-do-db = ["~^b.*","s1"] 
+#[[replicate-do-table]]
+#db-name ="test"
```
Comment (GregoryIan) [inline syncer/config.toml]: add a regex example

**Comment by GregoryIan** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**